libs/core/langchain_core/runnables/utils.py

```diff
@@ -748,7 +748,7 @@ def is_async_generator(
     """
     return (
         inspect.isasyncgenfunction(func)
-        or hasattr(func, "__call__")
```

Comment:
I think this was on purpose cc @hinthornw I think

Reply:
It comes from https://docs.astral.sh/ruff/rules/unreliable-callable-check/
Is it a false positive ? Should the rule be ignored here ?

Reply:
Here this is explicitly just a guard so that we don't err when checking `inspect.isasyncgenfunction(func.__call__)` in the next line, so i think it's a false positive.

Reply:
could alternatiely do `inspect.isasyncgenfunction(getattr(func, '__call__', None))` if we want the linter to be happy